homeassistant/components/axis/config_flow.py

```diff
@@ -156,7 +156,7 @@ async def async_step_dhcp(self, discovery_info: dict):
         return await self._process_discovered_device(
             {
                 CONF_HOST: discovery_info[IP_ADDRESS],
-                CONF_MAC: format_mac(discovery_info.get(MAC_ADDRESS)),
+                CONF_MAC: format_mac(str(discovery_info.get(MAC_ADDRESS))),
```

Comment:
```suggestion
                CONF_MAC: format_mac(discovery_info.get(MAC_ADDRESS, "")),
```
If you want to make sure to input a string to format_Mac. I'd like an explanation about what you want to do since please mypy isn't really a motivation

Reply:
Actually it is just allow axis to be monitored by mypy. However I agree with @frenck that casting None to a str is not the best way (apart from the fact that it happens to work).

Reply:
I haven't delved into the hole mypy thing. Is it only to verify typing?

Reply:
Yes it is to ensure that typing is correct, where used ! Using typing is a recommendation and not a demand.

I am not convinced that mypy ensures our code get better or more stable, but that is another discussion, since we use mypy, I agree that we should over time get all integrations controlled. Axis was one of many (roughly 1/3 of the total) that is excluded..and with this PR it is included.

And I am simply using a bit of my spare time, to help out.

Reply:
I looked at the code, this fix should work. `format_mac` will return original string if can't recognise the format. Then `_process_discovered_device` return abort on `self.async_abort(reason="not_axis_device")`.